components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/RuntimeArtifactGeneratorUtil.java

```diff
@@ -69,10 +69,18 @@ public static RuntimeArtifactDto generateMetadataArtifact(String apiId, String n
             throws APIManagementException {
 
         List<APIRuntimeArtifactDto> gatewayArtifacts = getRuntimeArtifacts(apiId, gatewayLabel, tenantDomain);
+        Map<String, String> contextUuidMap = new HashMap<>();
         if (gatewayArtifacts != null) {
+
+            if (StringUtils.isNotEmpty(apiId)) {
+                contextUuidMap.put(apiId, gatewayArtifactsMgtDAO.retrieveAPIContextFromApiId(apiId));
+            } else {
+                contextUuidMap = gatewayArtifactsMgtDAO.retrieveAllAPIContext();
```

Comment:
Do we need to get all API's contexts from here? Can we pass the same parameters apiId, gatewayLabel, organization to a method and retrieve a map of APIInfo objects?

Reply:
With this particular rest API we do not use the organization query as a filter.